cob_generic_can/common/src/SocketCan.cpp

```diff
@@ -30,174 +52,154 @@
+    if (m_bInitialized == false)
+    {
+        return false;
+    }
 
-    // wait until msg in buffer
+    can::Frame frame;
     bool bRet = false;
-    int i=0;
+    int i = 0;
+
     do
     {
-        if (recived) {
-            pCMsg->setID(recived_frame.id);
-            pCMsg->setLength(recived_frame.dlc);
-            pCMsg->set(recived_frame.data[0], recived_frame.data[1], recived_frame.data[2], recived_frame.data[3],
-                            recived_frame.data[4], recived_frame.data[5], recived_frame.data[6], recived_frame.data[7]);
+        if (m_reader.read(&frame, boost::chrono::seconds(1)))
+        { 
+            pCMsg->setID(frame.id);
+            pCMsg->setLength(frame.dlc);
+            pCMsg->set(frame.data[0], frame.data[1], frame.data[2], frame.data[3],
+                       frame.data[4], frame.data[5], frame.data[6], frame.data[7]);
             bRet = true;
             break;
         }
         i++;
-        usleep(10000);
+        usleep(1000);
     }
-    while(i < iNrOfRetry);
+    while ((i < iNrOfRetry && bRet != true));
     return bRet;
 }
 
 //-------------------------------------------
 bool SocketCan::receiveMsgTimeout(CanMsg* pCMsg, int nSecTimeout)
 {
-    if (m_bInitialized == false) return false;
+    if (m_bInitialized == false)
+    {
+        return false;
+    }
 
-    // wait until msg in buffer
     bool bRet = false;
+    can::Frame frame;
+    usleep(nSecTimeout / 1000);
 
-    usleep(nSecTimeout/1000);
-    if (recived) {
-        pCMsg->setID(recived_frame.id);
-        pCMsg->setLength(recived_frame.dlc);
-        pCMsg->set(recived_frame.data[0], recived_frame.data[1], recived_frame.data[2], recived_frame.data[3],
-                        recived_frame.data[4], recived_frame.data[5], recived_frame.data[6], recived_frame.data[7]);
+    if (m_reader.read(&frame, boost::chrono::seconds(1)))
+    {
+        pCMsg->setID(frame.id);
+        pCMsg->setLength(frame.dlc);
+        pCMsg->set(frame.data[0], frame.data[1], frame.data[2], frame.data[3], frame.data[4], frame.data[5], frame.data[6], frame.data[7]);
         bRet = true;
     }
-
     return bRet;
 }
 
-bool SocketCan::initCAN() {
-	m_bInitialized = true;
-        bool bRet = true;
-        return true;
-}
-
-void SocketCan::recive_frame(const can::Frame& frame){
-
-    if(frame.is_error){
-        std::cout << "E " << std::hex << frame.id << std::dec;
-    }else if(frame.is_extended){
-        std::cout << "e " << std::hex << frame.id << std::dec;
-    }else{
-        std::cout << "s " << std::hex << frame.id << std::dec;
-    }
-
-    std::cout << "\t";
-
-    if(frame.is_rtr){
-        std::cout << "r";
-    }else{
-        std::cout << (int) frame.dlc << std::hex;
-
-        for(int i=0; i < frame.dlc; ++i){
-            std::cout << std::hex << " " << (int) frame.data[i];
-        }
-    }
-
-    std::cout << std::dec << std::endl;
-    recived_frame = frame;
-    recived = true;
+bool SocketCan::initCAN()
```

Comment:
Oh yes now I know why I did it. Other nodes which are using this library are switching between CANPeakSysUSB and SocketCAN. That is the main reason to keep this. But I will take care to update them. This note could be also important for you too :)